ui/api/src/main/java/org/jboss/forge/addon/ui/annotation/AnnotatedCommandAggregator.java

```diff
@@ -0,0 +1,18 @@
+package org.jboss.forge.addon.ui.annotation;
+
+import org.jboss.forge.furnace.addons.AddonId;
+
+/**
+ * Class aggregating information about all the commands registered by annotation 
+ * @author mbriskar
+ *
+ */
+public interface AnnotatedCommandAggregator
+{
+
+   /**
+    * Clean all the commands aggregated for the destroyed addonId
+    * @param addonId
+    */
+   public void addonDestroyed(AddonId addonId);
```

Comment:
Maybe we could move this method to CommandFactory / CommandProvider. Not sure I like having this interface